Fix jira URL prefixes to work with new jira boards

tessi opened the pull request:
The JIRA board URLS changed for new JIRA boards. They now contain
optional trailing slashes and also allow further nested routes.
Examples:

* `/jira/software/projects/ABCD/boards/7`
* `/jira/software/projects/ABCD/boards/7/backlog`

Where before, only this URL was recognized:

* `/jira/software/projects/BCHP/boards/7/`

Closes #195.

tessi commented on src/common/adapters/jira.js:
```diff
@@ -19,7 +19,10 @@ function isJiraPage(loc, doc) {
 
-const pathSuffixes = new RegExp('/(browse/[^/]+|projects/[^/]+/issues/[^/]+|secure/RapidBoard.jspa|jira/software/projects/[^/]+/boards/[^/]+/)$', 'g');
+const pathSuffixes = new RegExp(
+  '/(browse/[^/]+|projects/[^/]+/issues/[^/]+|secure/RapidBoard.jspa|jira/software/projects/[^/]+/boards/.*)$',
```
`jira/software/projects/[^/]+/boards/.*` captures all possible trailing URL parts of the board view. I opted to not be very specific here and just catch all possible routes within `/boards/` to be more stable against further JIRA changes.

pmeinhardt replied:
Cool. This should also cover what we had here: https://github.com/bitcrowd/tickety-tick/pull/216/files#diff-a4549fad9112b44c628079812d266381L22-R22

pmeinhardt commented:
Is this a different problem from: https://github.com/bitcrowd/tickety-tick/pull/216?

tessi commented:
> Is this a different problem from: #216?

This is the exact issue i fixed - by not scanning (and removing) the prefix path correctly that weird URL from #216 gets built.

tessi commented:
@pmeinhardt true that, added tests. Actually one of the existing tests passed although the wrong jira base_url was called. I added a check against which API URL is called to the relevant tests.
